### ISSUE TITLE
{% static %} doesn't include hash

### ISSUE THREAD

**Issue by MaZZly** [evansd/whitenoise#43]
Can't get the darn thing to work with 
`STATICFILES_STORAGE = 'whitenoise.django.GzipManifestStaticFilesStorage'`

But with my own homebrewed combination of django-pipeline and whitenoise:
```
from pipeline.storage import PipelineMixin
from whitenoise.django import GzipManifestStaticFilesStorage

class GzipManifestPipelineStorage(PipelineMixin, GzipManifestStaticFilesStorage):
    pass
```
and then 
`STATICFILES_STORAGE = 'app.storages.GzipManifestPipelineStorage'`
at least the {% compressed_css %} tag of django-pipeline outputs the hash. but {%static%} still doesn't

So tl;dr: Following documentation the {%static%} tag outputs style.css instead of style.hash.css

**Comment by evansd** [comment]
It could be that you're missing the `{% load static %}` declaration at the
top of your template file.

Can't get the darn thing to work with
STATICFILES_STORAGE = 'whitenoise.django.GzipManifestStaticFilesStorage'

But with my own homebrewed combination of django-pipeline and whitenoise:

from pipeline.storage import PipelineMixin
from whitenoise.django import GzipManifestStaticFilesStorage

class GzipManifestPipelineStorage(PipelineMixin,
GzipManifestStaticFilesStorage):
    pass

STATICFILES_STORAGE = GzipManifestPipelineStorage

at least the {% compressed_css %} tag of django-pipeline seems to output
the hash. ({%static%} still doesn't)

So tl;dr: Following documentation the {%static%} tag outputs style.css
instead of style.hash.css

—
Reply to this email directly or view it on GitHub
<https://github.com/evansd/whitenoise/issues/43>.

**Comment by RyanBalfanz** [comment]
@evansd Should we `{% load static %}` or `{% load static from staticfiles %}` as in [the docs](http://whitenoise.evans.io/en/latest/django.html#make-sure-staticfiles-is-configured-correctly)?

It seems like different suggestions exists from different authorities -- If one doesn't read the [WhiteNoise Docs](http://whitenoise.evans.io/en/latest/index.html) and just follows Django's [static files how-to](https://docs.djangoproject.com/en/1.8/howto/static-files/) and Heroku's [django-assets](https://devcenter.heroku.com/articles/django-assets) article they wouldn't know to prefer the latter. It's a bit unclear IMO.

**Comment by evansd** [comment]
@RyanBalfanz your comment seems to have disappeared from github, so I'm copying here for reference:

> @evansd Should we {% load static %} or {% load static from staticfiles %} as in the docs?
>
> It seems like different suggestions exists from different authorities -- If one doesn't read the WhiteNoise Docs and just follows Django's static files how-to and Heroku's django-assets article they wouldn't know to prefer the latter. It's a bit unclear IMO.

Sorry, that was a typo. It should be `{% load staticfiles %}` or `{% load static from staticfiles %}`. Both do exactly the same thing but the latter is a bit more explicit about which tag is being imported.

It's a confusing feature of Django that the `static` tag behaves differently depending on whether you've loaded the `staticfiles` tag library or not. But for backwards compatibility reasons that isn't going to change so we just have to work with it.

I think the Heroku docs should be amended to mention that you should load the `staticfiles` library in your templates. Do you mind making that suggestion to them? There's a feedback form on the docs page.

**Comment by evansd** [comment]
Assuming this resolved the issue.
